src/core/layertree/qgslayertreemodellegendnode.cpp

```diff
@@ -604,7 +619,94 @@ void QgsSymbolLegendNode::updateLabel()
   emit dataChanged();
 }
 
+QString QgsSymbolLegendNode::evaluateLabel( QgsExpressionContext context, QString label )
```

Comment:
Help ensure that the label is properly converted as an expression in order to be evaluated